state-validate-util/src/main/java/org/sunbird/statevalidateutil/RecordProcessor.java

```diff
@@ -0,0 +1,206 @@
+                                endTracingRecord(userObject.getUserId());
+                            }
+                        });
+
+        connection.closeConnection();
+        closeWriterConnection();
+    }
+
+    private int calcFieldsValue(Map<String, Object> userBooleanMap) {
+        int userFlagValue = 0;
+        Set<Map.Entry<String, Object>> mapEntry = userBooleanMap.entrySet();
+        for(Map.Entry<String, Object> entry: mapEntry) {
+            if(StringUtils.isNotEmpty(entry.getKey()) && (Boolean) entry.getValue()) {
+                userFlagValue += boolFlagValue(entry.getKey());
+            }
+        }
+        return userFlagValue;
+    }
+
+    public int boolFlagValue(String userFlagType) {
+        int decimalValue = 0;
+        if(userFlagType.equals(UserBoolFlagEnum.PHONE_VERIFIED.getUserFlagType())) {
+            decimalValue = UserBoolFlagEnum.PHONE_VERIFIED.getUserFlagValue();
+        } else if (userFlagType.equals(UserBoolFlagEnum.EMAIL_VERIFIED.getUserFlagType())) {
+            decimalValue = UserBoolFlagEnum.EMAIL_VERIFIED.getUserFlagValue();
+        } else if (userFlagType.equals(UserBoolFlagEnum.STATE_VALIDATED.getUserFlagType())) {
+            decimalValue = UserBoolFlagEnum.STATE_VALIDATED.getUserFlagValue();
+        }
+        return decimalValue;
+    }
+
+    private boolean isStateUSer(String userId, String custodianOrgId) {
+        String orgId = null;
+        ResultSet resultSet = connection.getRecords("select organisationid from user_org where userid='"+userId+"'");
+        Iterator<Row> iterator = resultSet.iterator();
+        Row row = iterator.next();
+        orgId = row.getString(0);
+        if(!orgId.equals(custodianOrgId)) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private String getSystemSettingCustodianOrgId() {
+        String custOrgId = null;
+        ResultSet resultSet = connection.getRecords("select value from system_settings where field='custodianOrgId'");
+        Iterator<Row> iterator = resultSet.iterator();
+        Row row = iterator.next();
+        custOrgId = row.getString(0);
+        return custOrgId;
+    }
+
+    private static Map<String, String> getCompositeKeysMap(User user) {
+        Map<String, String> compositeKeysMap = new HashMap<>();
+        compositeKeysMap.put(DbColumnConstants.userId, user.getUserId());
+        return compositeKeysMap;
+    }
+
+    /**
+     * this method will perform sequential operation of db records
+     * - generate insert query
+     * - decrypt the externalIds and originalExternalIds
+     * - insert record
+     * - insert Record passes will then delete the record...
+     *
+     * @param user
+     * @param compositeKeysMap
+     */
+    private void performSequentialOperationOnRecord(User user, Map<String, String> compositeKeysMap, boolean isStateValidated) {
+
+        String query = CassandraHelper.getInsertRecordQuery(user);
```

Comment:
we are doing operation with existing user , so it should be updateQuery not getInsertRecordQuery

Reply:
changed to getUpdateRecordQuery